src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/measure/ActionMeter.kt

```diff
@@ -76,50 +97,80 @@ class ActionMeter(
         observation: (T) -> JsonObject?
     ): T {
         val start = clock.instant()
+        var actionResult: ActionResult = ActionResult.OK
```

Comment:
We avoid `var`s whenever possible. Tracking mutable state is error-prone. Builders already encapsulate mutability.
So I'd start with an `ActionMetric.Builder`, filling in the common fields, like `label`, `start` and `virtualUser` and let the rest of the control flow contribute to the state.

- [ ] avoid non-builder `var`s